python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1529,7 +1529,7 @@ def _execute_asset_backfill_iteration_inner(
         candidate_asset_graph_subset = _get_candidate_asset_graph_subset(
             asset_backfill_data,
             asset_graph_view,
-            materialized_since_last_tick,
+            updated_materialized_subset,
```

Comment:
what's the reasoning for this change? this makes it so we re-evaluate all assets that have a parent that has ever been materialized every tick, which in theory is somewhat unrelated to the other changes.

ideally, this would be fine (i.e. one possibility if that we just evaluate everything all the time), but there's some rough edge cases in the self-dependent asset tests that made me a bit scared of that sort of change.

Reply:
to add a bit more color (as I just looked at bk and saw that a similar test was failing), the core issue here is that the code for self-dependent assets _implicitly depends on_ the fact that we only evaluate the asset when one of its parents has updated this tick.

this is because our logic doesn't actually currently have any provisions forcing the upstream partition to be materialized once it's being evaluated. so if we evaluate the whole self-dependent asset every time, each evaluation will emit a batch of partitions for the self-dependent asset.